hta/common/execution_trace.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import gzip
+import json
+import logging
+import sys
+import time
+
+from typing import List  # Dict, , NamedTuple, Optional
+
+import numpy as np
+import pandas as pd
+
+from hta.common.trace import Trace
+from hta.configs.config import logger
+from hta.utils.utils import normalize_path
+
+from param.python.tools.execution_graph import ExecutionGraph
+
+# PyTorch Events types that are correlated in the Execution Trace
+EXECUTION_TRACE_SUPPORTED_EVENTS: List[str] = ["cpu_op", "user_annotation"]
+
+
+def load_execution_trace(et_file: str) -> ExecutionGraph:
+    """Loads Execution Trace from json file and parses it into an
+    object representation. For large files this could take a lot of memory.
+
+    Args:
+        et_file (str): File path for the Execution Trace.
+
+    Returns:
+        ExecutionGraph object.
+    """
+    et_file_path = normalize_path(et_file)
+
+    t_start = time.perf_counter()
+    with gzip.open(et_file_path, "rb") if et_file.endswith("gz") else open(
+        et_file_path, "r"
+    ) as f:
+        et = ExecutionGraph(json.load(f))
+    t_end = time.perf_counter()
+
+    t_end = time.perf_counter()
+    logger.info(
+        f"Parsed Execution Trace file {et_file}, "
+        f"time = {(t_end - t_start):.2f} seconds "
+    )
+    return et
+
+
+def _et_has_overlap(trace_df: pd.DataFrame, et: ExecutionGraph) -> bool:
+    """Use record function IDs (rf_id) to find out if ET and Kineto trace
+    have overlap"""
```

Comment:
Please follow the same documentation style as in the function above.

Reply:
I thought this is an internal function hence didn't let me fix this